### PR TITLE
Show the correct homepage when previewing a theme in the Site Editor

### PR DESCRIPTION
**Pull request by okmttdhr**
## Proposed changes:

Show the correct homepage when previewing a theme in the Site Editor to support [Block Theme Previews](https://github.com/WordPress/gutenberg/pull/50030) on dotcom. This is done by overriding the `show_on_front` option to `posts`.

I added changes in ~~Jetpack~~ jetpack-mu-wpcom since dotcom wants to support Block Theme Previews both on Simple and Atomic sites. 

This PR is part of https://github.com/Automattic/wp-calypso/issues/79221.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

Atomic sites

* Create an Atomic site
* Change your Reading Settings to show a static page as a homepage
* Patch this PR to your site by following https://github.com/Automattic/jetpack/pull/31929#issuecomment-1639638285
	* Note that you must set `define( 'JETPACK_MU_WPCOM_LOAD_VIA_BETA_PLUGIN', true );` to your `wp-config.php`
* Install a theme 
* Go to Block Theme Previews with `/wp-admin/site-editor.php?wp_theme_preview=<your-theme>` (e.g. `?wp_theme_preview=pendant`)
* Verify Block Theme Previews show Index or Home template as a homepage

Simple sites

- Create a site
- Change your Reading Settings to show a static page as a homepage
- Patch this PR to your sandbox by following https://github.com/Automattic/jetpack/pull/31929#issuecomment-1639638285
- Sandbox your testing site AND `public-api.wordpress.com`
- Go to Block Theme Previews with `/wp-admin/site-editor.php?wp_theme_preview=<your-theme>` (e.g. `?wp_theme_preview=pub/pendant`)
- Verify Block Theme Previews show Index or Home template as a homepage

self-hosted site

- Patch this PR to your site
	- I used `jetpack rsync` PCYsg-eg0-p2
- Go to Block Theme Previews with `/wp-admin/site-editor.php?wp_theme_preview=<your-installed-theme>` (e.g. `?wp_theme_preview=twentytwentytwo`)
- Verify nothing breaks on the Site Editor

### CODE REVIEW

**Review by taipeicoder — DISMISSED** [review]
Testes:

Atomic sites: ✅  BTP shows Index of Home template as homepage.
Simple sites: ✅ BTP works as before.
Self-hosted sites: ✅ BTP works as before.

**Review by jeherve — COMMENTED** [review]
I'm not sure I fully understand why this is in Jetpack. I think I'm missing some context on this, probably. 

If we always want to show the posts page when previewing a new theme, shouldn't we make that change in Core directly, since theme previews will soon be available to everyone?

**Comment by arthur791004** [comment]
@jeherve Based on pekYwv-1BS-p2 and pekYwv-1BS-p2#comment-1187, I guess Core won't touch the reading settings in the short term since Core won't change the reading settings during the theme switch. So, it makes sense to keep the current reading settings when you're previewing a theme in the site editor on Core.

However, dotcom has a mechanism to update the reading settings based on the headstart annotation of the theme. So, we have to override the `show_on_front` option dynamically on both simple and atomic sites to make the preview look the same after the theme switch. That's why we made the change on Jetpack.

Does it make sense? Or any thoughts?

**Comment by jeherve** [comment]
Ok, that makes sense. In this situation, and since this is really only interesting for WordPress.com sites (simple and WoA), maybe this should be shipped in the mu-wpcom plugin instead?
https://github.com/Automattic/jetpack/tree/trunk/projects/plugins/mu-wpcom-plugin

**Comment by arthur791004** [comment]
Oh yes, that makes sense. Thank you 👍

**Comment by okmttdhr** [comment]
Thank you for your comment, @jeherve!
I was aware of mu-wpcom-plugin, but Gutenberg itself is not a plugin that users must use, so I thought putting the code in there was a bit weird.
But that makes sense if we consider this as a feature that wpcom provides for both Simple and Atomic sites. I'll address this!

And thank you so much for your detailed explanation, @arthur791004!

File: projects/packages/jetpack-mu-wpcom/src/features/block-theme-previews/block-theme-previews.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+/**
+ * Always show the correct homepage when previewing a theme in the Site Editor
+ *
+ * @see https://github.com/Automattic/wp-calypso/issues/79221
+ * @since 12.4
+ */
+add_filter(
```
Comment (arthur791004) [inline projects/packages/jetpack-mu-wpcom/src/features/block-theme-previews/block-theme-previews.php:14]: Maybe we can add the filter or even load this feature only when the `wp_theme_preview` is set 🤔

Comment (okmttdhr) [reply projects/packages/jetpack-mu-wpcom/src/features/block-theme-previews/block-theme-previews.php:14]: Addressed in https://github.com/Automattic/jetpack/pull/31929/commits/3800d38442b703333f764cf5559a17c0d9d4ddae!

**Review by arthur791004 — DISMISSED** [review]
👍

**Comment by okmttdhr** [comment]
@Automattic/lego I requested your reviews again as I resolved conflicts 🙏 Thank you!

**Review by fushar — COMMENTED** [review]
Tested by using the following scenarios:

- current theme: Appleton, which has a static homepage set.
- previewing theme: Pendant, which has the Home template.

### Atomic site ✅ 

|Before|After|
|-|-|
|image|image|

### Simple site ❌ 

|Before|After|
|-|-|
|image|image|

I am not sure what I'm doing wrong. I have sandboxed my site (see the "Sandboxed" badge at the left-corner of the right screenshot). Log:

```
$ bin/jetpack-downloader test jetpack-mu-wpcom-plugin add/site-editor-homepage
Current checkout is switch-theme-logic-new. Check out trunk? [Y/n] Y

Updating trunk...
# Request received by "phabricator-b06-29.dfw.automattic.com", forwarding to cluster host "phabricator-b12-10.dfw.automattic.com".
# Acquiring read lock for repository "wpcom-git" on device "phabricator-b12-10.dfw.automattic.com"...
# Acquired read lock immediately.
# Device "phabricator-b12-10.dfw.automattic.com" is already a cluster leader and does not need to be synchronized.
# Cleared to fetch on cluster host "phabricator-b12-10.dfw.automattic.com".

Updating files:  53% (12198/23014)
Updating files:  61% (14039/23014)
Updating files: 100% (23014/23014), done.
Determining jetpack-mu-wpcom-plugin download URL...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 10406  100 10406    0     0  47733      0 --:--:-- --:--:-- --:--:-- 47733
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8181  100  8181    0     0  71763      0 --:--:-- --:--:-- --:--:-- 72398
Downloading jetpack-mu-wpcom-plugin...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 80835  100 80835    0     0   400k      0 --:--:-- --:--:-- --:--:--  398k
Unpacking jetpack-mu-wpcom-plugin...
jetpack-mu-wpcom-plugin add/site-editor-homepage is now checked out in sun!
If you intend to commit it, use `git add -Af wp-content/mu-plugins/jetpack-mu-wpcom-plugin/sun` to ensure all files are added.
To reset your environment instead, use `jetpack-downloader reset jetpack-mu-wpcom-plugin`.
```

---

Will test on a JN site soon.

**Comment by okmttdhr** [comment]
>I am not sure what I'm doing wrong. I have sandboxed my site (see the "Sandboxed" badge at the left-corner of the right screenshot). 

This is weird. 👀 I will check it Monday!
One thing I want to confirm is if you sandboxed `public-api.wordpress.com` as well. @fushar

**Comment by fushar** [comment]
> Will test on a JN site soon.

### Jetpack-connected self-hosted site ✅ 

Nothing breaks!

**Comment by taipeicoder** [comment]
Gave a test on Simple Sites, and it works as expected 👍

**Review by fushar — APPROVED** [review]
Update:

### Simple site ✅ 

|Before|After|
|-|-|
|image|image|

Needed to change `.config/jetpack-loader.php` from `moon` to `sun`.

**Pull request merged**